src/run/run.c

```diff
@@ -1848,7 +1848,8 @@ static int start_transient_service(sd_bus *bus) {
                         if (!isempty(arg_background))
                                 (void) pty_forward_set_background_color(c.forward, arg_background);
 
-                        set_window_title(c.forward);
+                        if (shall_set_terminal_title())
+                                set_window_title(c.forward);
```

Comment:
the invocations of pty_forward_set_title_prefix() should also be conditionalized like this.

And I think we should check for the env var generically in ptyfwd code, not in the individual tools such as systemd-run.

i.e. move this check into insert_window_title_fix(), pty_forward_disconnect(), pty_forward_ansi_process(), do_shovel() instead, i.e. whenever we are about to use the title or title_prefix field.

Reply:
(and you are right, we don't do that for background tinting right now. And yes, we should actually do that too for that)

Reply:
> the invocations of pty_forward_set_title_prefix() should also be conditionalized like this.

I don't see any invocations of it except within set_window_title()? Do you mean the change in ee0a085ab2fa8f152319d460334d679f822fadb2? (Though I think making functions become no-op from an envvar kind of reduces clarity, not improves it...)

> And I think we should check for the env var generically in ptyfwd code, not in the individual tools such as systemd-run.

Hmm, in earlier revisions I deliberately moved it out of ptyfwd to allow for different defaults between tools (i.e. making it off-by-default in systemd-run while remaining on-by-default in run0)...

Reply:
Yeah, as mentioned I think it makes sense to disable this by default for systemd-run. Plus we have a rule for not hvaing negatively-named envvar specifically to allow flipping the defaults.